internal/fields/validate.go

```diff
@@ -302,6 +302,21 @@ func FindElementDefinition(searchedKey string, fieldDefinitions []FieldDefinitio
 	return findElementDefinitionForRoot("", searchedKey, fieldDefinitions)
 }
 
+var compileCache map[string]*regexp.Regexp
+
+func init() {
+	compileCache = make(map[string]*regexp.Regexp)
+}
+
+func mustCompile(str string) *regexp.Regexp {
```

Comment:
nit: in elastic-package, we used to place such hacks or extensions in corresponding packages with -ext suffix, so `regexpext`. You can consider applying it here too.